assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,316 @@
+            } catch (e) {
+              return;
+            }
+          }
+        }, 500);
+
+        // Check if the editor isn't empty to disable the 'Add' button. Also check in what format the input is in (JSON/YAML) and change
+        // the editor accordingly.
+        $scope.aceChanged = function() {
+          aceDebounce();
+        };
+
+        $scope.create = function() {
+          delete $scope.alerts['create'];
+          delete $scope.alerts['parsing'];
+          var resource;
+
+          // Trying to auto-detect what format the input is in. Since parsing JSON throws only SyntexError
+          // exception if the string to parse is not valid JSON, it is tried first and then the YAML parser
+          // is trying to parse the string. If that fails it will print the reason. In case the real reason
+          // is JSON related the printed reason will be "Reason: Unable to parse", in case of YAML related
+          // reason the true reason will be printed, since YAML parser throws an error object with needed
+          // data.
+          try {
+            resource = JSON.parse($scope.editorContent);
+          } catch (e) {
+            try {
+              resource = YAML.parse($scope.editorContent);
+            } catch (e) {
+              $scope.error = e;
+              return;
+            }
+          }
+
+          if (!resource.metadata || !resource.kind) {
+            missingFieldsError(resource);
+            return;
+          }
+
+          $scope.resourceKind = resource.kind;
+
+          if ($scope.resourceKind === "Template") {
+            $scope.templateOptions = {
+              process: true,
+              add: true
+            };
+          }
+
+          if ($scope.resourceKind.endsWith("List")) {
+            $scope.isList = true;
+          }
+          
+          if ($scope.isList) {
+            $scope.resourceList = resource.items;
+            $scope.resourceName = '';
+          } else {
+            $scope.resourceList = [resource];
+            $scope.resourceName = resource.metadata.name;
+          }
+
+          $scope.updateResources = [];
+          $scope.createResources = [];
+
+          var resourceCheckPromises = [];
+          var errorOccured = false;
+          _.forEach($scope.resourceList, function(item) {
+            if (!item.metadata || !item.kind) {
+              missingFieldsError(item);
+              errorOccured = true;
+              return false;
+            } else if (item.metadata.namespace && item.metadata.namespace !== $scope.projectName) {
+              $scope.error = {
+                message: item.kind + " " + item.metadata.name + " can't be created in project " + item.metadata.namespace + ". Can't create resource in different projects."
+              };
+              errorOccured = true;
+              return false;
+            }
+            resourceCheckPromises.push(checkResource(item));
+          });
+
+          if (!errorOccured) {
```

Comment:
I find it easier to understand

```javascript
if (errorOccured) {
  return;
}

$q.all(...).then(...);
```

Then if someone adds something else here, they won't make the mistake of adding it outside the `if`.